File: VoceViuWeb/Models/Authentication/AuthenticateRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using VoceViuModel.Users.Commands;

namespace VoceViuWeb.Models.Authentication
{
    public class AuthenticateRequest : AuthenticateMessage
    {
        public string ReturnUrl { get; set; }

        public IEnumerable<string> Validate()
        {
            if (String.IsNullOrWhiteSpace(Username))
                yield return "Deve-se fornecer um usuário";

            if (String.IsNullOrWhiteSpace(Password))
                yield return "Deve-se fornecer uma senha";
        }
    }
}

File: VoceViuWeb/Scripts/App/Models/AuthenticateRequest.js
var AuthenticateRequest = function () {
    this.Username = "";
    this.Password = "";
    this.RememberMe = false;

    var self = this;

    this.Validate = function () {

        var errors = [];

        if (!self.Username)
            errors.push("Deve-se informar o usuário");

        if (!self.Password)
            errors.push("Deve-se informar a senha");

        return errors;
    };
};